extension/src/background/messageListener/freighterApiMessageListener.ts

```diff
@@ -95,282 +99,347 @@ export const freighterApiMessageListener = (
           resolve({ publicKey: publicKeySelector(sessionStore.getState()) });
         }
 
-        resolve({ error: "User declined access" });
+        resolve({
```

Comment:
This file has a huge diff but the main changes here are:

1. Returning 2 kinds of errors to maintain backwards compat
2. Wrapping all these methods in try...catch so on any type of runtime error, we can surface an error